src/VirtoCommerce.ExperienceApiModule.DigitalCatalog/Schemas/DigitalCatalogSchema.cs

```diff
@@ -56,7 +56,7 @@ public void Build(ISchema schema)
                     //TODO:  Need to check what there is no any alternative way to access to the original request arguments in sub selection
                     context.CopyArgumentsToUserContext();
                     var loader = _dataLoader.Context.GetOrAddBatchLoader<string, ExpProduct>("productsLoader", (ids) => LoadProductsAsync(_mediator, ids, context));
-                    return await loader.LoadAsync(context.GetArgument<string>("id"));
+                    return await loader.LoadAsync(context.GetArgument<string>("id")).GetResultAsync();
```

Comment:
Can you split this line into two to prevent possible deadlocks?

Reply:
done